### PR TITLE
Unify value proto between daml-lf and ledger-api

### PR DESCRIPTION
**Pull request by mziolekda**
- align the field numbers
- align the field names
- change contract-id type from string to bytes

Field nesting still cold be improved.

### CODE REVIEW

File: sdk/daml-lf/ledger-api-value/src/main/protobuf/com/daml/ledger/api/v2/value.proto [hunk 1]
```diff
@@ -59,97 +75,80 @@ message Value {
     string text = 8;
 
     // Identifier of an on-ledger contract. Commands which reference an unknown or already archived contract ID will fail.
-    // Must be a valid LedgerString.
-    string contract_id = 9;
+    // Must encode a valid LedgerString.
+    bytes contract_id = 9;
```
Comment (mziolekda) [inline sdk/daml-lf/ledger-api-value/src/main/protobuf/com/daml/ledger/api/v2/value.proto:79]: This is the most significant change in this PR. The contract-id changes type to `bytes`

Comment (mziolekda) [reply sdk/daml-lf/ledger-api-value/src/main/protobuf/com/daml/ledger/api/v2/value.proto:79]: @remyhaemmerle-da 
I get a lot of questions about contract_id being bytes. I thought there was some deep reason for that related to canton protocol. But there doesn't seem to be one. Could we go in the opposite direction and redefine daml-lf's contract-id as string?

Comment (mziolekda) [reply sdk/daml-lf/ledger-api-value/src/main/protobuf/com/daml/ledger/api/v2/value.proto:79]: It used to be a ContractId message that contained a string. It became bytes  months ago:
https://github.com/digital-asset/daml/commit/8227412a6bb2ac4bf68a481ead81696154e1a0bf

Comment (remyhaemmerle-da) [reply sdk/daml-lf/ledger-api-value/src/main/protobuf/com/daml/ledger/api/v2/value.proto:79]: There is no deep reason about represented contract id as string expect 
- using a closer reprensantation of the engine internal
- saving space (as string they need 140bytes as bytestring we need 70bytes)

> It used to be a ContractId message that contained a string. It became bytes 6 months ago:

Comment (remyhaemmerle-da) [reply sdk/daml-lf/ledger-api-value/src/main/protobuf/com/daml/ledger/api/v2/value.proto:79]: If we want to revert back, we will need to either 
- add a new field for new LF version
- keep the bytes type and populate it with string.

**Comment by mziolekda** [comment]
We have discussed this extensively and decided against touching it:

- daml-lf values stored in the db become fossilized for a long time; because they are versioned, it is always possible to thaw them and convert to the contemporary version of ledger-api; it is not guaranteed though that the binary format will remain compatible across versions, so a naive reinterpretation of daml-lf proto with ledger-api bindings will not be universally possible; we will always need de- and en-coding to guarantee backwards compatibility
- there are good reasons for bytes string internal representation (space, close to speedy) and for hexString representation on the ledger-api (human-readable, json ready)

**Comment by mziolekda** [comment]
Closing

**Pull request closed without merging**